backend/src/common/files/file-upload-client.ts

```diff
@@ -55,23 +55,35 @@ class FileUploadClient implements IFileUploadClient {
     public async upload({
         buffer,
         contentType,
+        contentEncoding,
+        key,
     }: FileUploadRequestDto): Promise<FileUploadResponseDto> {
         const { AWS } = this.config.ENV;
 
-        const key = this.generateFileKey();
+        const fileKey = key ?? this.generateFileKey();
+
+        let body = buffer;
+
+        if (contentEncoding === ContentEncoding.BASE64) {
+            body = Buffer.from(
+                (buffer as string).replace(/^data:image\/\w+;base64,/, ''),
+                'base64',
```

Comment:
```suggestion
                ContentEncoding.BASE64,
```
`(buffer as string).replace(/^data:image\/\w+;base64,/, '')` - can you prettify this line so it can be more readable?